modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/Op.scala

```diff
@@ -0,0 +1,64 @@
+package ch.epfl.bluebrain.nexus.kg.indexing.filtering
+
+/**
+  * Enumeration type for all filtering operations.
+  */
+sealed trait Op extends Product with Serializable
+
+object Op {
+
+  /**
+    * Enumeration type for all filtering logical operators.
+    */
+  sealed trait LogicalOp extends Op
+  final case object And extends LogicalOp
+  final case object Or extends LogicalOp
+  final case object Not extends LogicalOp
+  final case object Xor extends LogicalOp
+
+  object LogicalOp {
+    /**
+      * Attempts to construct ''LogicalOp'' from the provided string.
+      *
+      * @param string the string representation of the logical operator
+      */
+    final def fromString(string: String): Option[LogicalOp] = string match {
```

Comment:
wouldnt `Try[LogicalOp]` as return type make more sense here? it's definitely an error when an incorrect operator is supplied

Reply:
In this particular case there's no additional information that an exception can provide. If the match succeeds the function will return the value, or return no value otherwise.